Automattic/wp-calypso
# Store: Add parent category selection to the management form

This PR adds the ability to select a parent category when updating or creating a new product category. It uses Calypso's `term-tree-selector` component.

Parent branch: #21347.
See: #20295.

To Test:
* Boot up this branch and go to `http://calypso.localhost:3000/store/products/categories/:site`
* Select a category that is a parent category
* Make sure that the "top level category" checkbox is shown
* Uncheck it, and make sure that the category selector appears
* Verify that you can select category as parent
* Go back to the list and select a sub-category
* Verify that its parent is marked in the selector component
* Go back once more to the list, and click "Add new category". Make sure the component works in the create context as well.

<img width="1147" alt="screen shot 2018-01-08 at 4 30 34 pm" src="https://user-images.githubusercontent.com/689165/34699645-690b949e-f493-11e7-8d46-bc6dbe81750f.png">


Note: Saving has not been hooked up yet.
